node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeCandidate.java

```diff
@@ -628,7 +628,9 @@ public ExclusivityViolation violatesExclusivity(ClusterSpec cluster, Application
 
             // this cluster requires exclusivity, but the parent is not exclusive
             if (exclusiveAllocation && parent.flatMap(Node::exclusiveToApplicationId).isEmpty())
-                return makeExclusive ? ExclusivityViolation.PARENT_HOST_NOT_EXCLUSIVE : ExclusivityViolation.YES;
+                return Preparer.requireParentHostLock(makeExclusive, type(), hostSharing) ?
```

Comment:
It's a bit weird asking whether you need a lock, to determine if exclusivity is violated, but I suppose it's otherwise code duplication. Perhaps rename the method, though?

Reply:
:+1: